src/components/ClusterNodes/NodeList.tsx

```diff
@@ -0,0 +1,526 @@
+                        _errorTitle = _errorTitle ? ', ' : '' + _nodeErrors.join(', ')
+                        for (let i = 0; i < _nodeErrors.length; i++) {
+                            const _errorLength = response[1].result.nodeErrors[_nodeErrors[i]].length
+                            _errorList.push(
+                                `${_nodeErrors[i]} on ${
+                                    _errorLength === 1 ? `${_errorLength} node` : `${_errorLength} nodes`
+                                }`,
+                            )
+                        }
+                    }
+                    setClusterErrorTitle(_errorTitle)
+                    setClusterErrorList(_errorList)
+                }
+                setLoader(false)
+            })
+            .catch((error) => {
+                showError(error)
+                setLoader(false)
+            })
+    }
+
+    useEffect(() => {
+        getNodeListData()
+    }, [clusterId])
+
+    useEffect(() => {
+        getClusterList()
+            .then((response: ClusterListResponse) => {
+                setLastDataSync(!lastDataSync)
+                if (response.result) {
+                    const optionList = response.result
+                        .filter((cluster) => !cluster.errorInNodeListing)
+                        .map((cluster) => {
+                            const _clusterId = cluster.id?.toString()
+                            if (_clusterId === clusterId) {
+                                setSelectedCluster({
+                                    label: cluster.name,
+                                    value: _clusterId,
+                                })
+                            }
+                            return {
+                                label: cluster.name,
+                                value: _clusterId,
+                            }
+                        })
+                    setClusterList(optionList)
+                }
+            })
+            .catch((error) => {
+                showError(error)
+            })
+    }, [])
+
+    useEffect(() => {
+        const _lastDataSyncTime = Date()
+        setLastDataSyncTimeString('Last refreshed ' + handleUTCTime(_lastDataSyncTime, true))
+        const interval = setInterval(() => {
+            setLastDataSyncTimeString('Last refreshed ' + handleUTCTime(_lastDataSyncTime, true))
+        }, 1000)
+        return () => {
+            clearInterval(interval)
+        }
+    }, [lastDataSync])
+
+    const handleFilterChanges = (): void => {
+        let _flattenNodeList = []
+        for (let index = 0; index < flattenNodeList.length; index++) {
+            const element = flattenNodeList[index]
+            if (selectedVersion.value !== defaultVersion.value && element['k8sVersion'] !== selectedVersion.value) {
+                continue
+            }
+            if (selectedSearchTextType === 'name' && element['name'].indexOf(searchText) === -1) {
```

Comment:
combine both if conditions with `||` and keep the else if as `if`

Reply:
having multiple if, else & loops in a single function increases the `Cyclomatic Complexity`.

Reply:
Done